sys/oom_linux_test.go

```diff
@@ -23,47 +23,63 @@ import (
 	"testing"
 	"time"
 
+	"github.com/containerd/containerd/pkg/userns"
 	"gotest.tools/v3/assert"
 	is "gotest.tools/v3/assert/cmp"
 )
 
 func TestSetPositiveOomScoreAdjustment(t *testing.T) {
+	// Setting a *positive* OOM score adjust does not require privileged
 	_, adjustment, err := adjustOom(123)
-	if err != nil {
-		t.Error(err)
-		return
-	}
+	assert.NilError(t, err)
 	assert.Check(t, is.Equal(adjustment, 123))
 }
 
 func TestSetNegativeOomScoreAdjustmentWhenPrivileged(t *testing.T) {
-	if RunningUnprivileged() {
-		t.Skip("Needs to be run as root")
+	if !runningPrivileged() || userns.RunningInUserNS() {
+		t.Skip("requires root and not running in user namespace")
 		return
 	}
 
 	_, adjustment, err := adjustOom(-123)
-	if err != nil {
-		t.Error(err)
-		return
-	}
+	assert.NilError(t, err)
 	assert.Check(t, is.Equal(adjustment, -123))
 }
 
 func TestSetNegativeOomScoreAdjustmentWhenUnprivilegedHasNoEffect(t *testing.T) {
-	if RunningPrivileged() {
-		t.Skip("Needs to be run as non-root")
+	if runningPrivileged() && !userns.RunningInUserNS() {
+		t.Skip("needs to be run as non-root or in user namespace")
 		return
 	}
 
 	initial, adjustment, err := adjustOom(-123)
-	if err != nil {
-		t.Error(err)
-		return
-	}
+	assert.NilError(t, err)
 	assert.Check(t, is.Equal(adjustment, initial))
 }
 
+func TestSetOOMScoreBoundaries(t *testing.T) {
```

Comment:
this test doesn't use any of the constants and hardcodes everything. It seems highly unlikely there will be churn in the kernel on the range, but you *could* use the constants here and be future-proof :)

Reply:
I went back-and-forth over this one as both ways made sense;
train of thought was that I like tests to use a "fixture" so that in case things change in code, tests don't "implicitly" evolve; that way (hopefully) catching change in behavior instead of "being updated automatically".

I think in this case, chances are _very_ low that this one will ever change, so let me update to use the consts

Reply:
Updated